presto-hive/src/main/java/com/facebook/presto/hive/StoragePartitionLoader.java

```diff
@@ -346,8 +348,23 @@ private static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inpu
                             partitionName));
         }
 
-        // Sort FileStatus objects (instead of, e.g., fileStatus.getPath().toString). This matches org.apache.hadoop.hive.ql.metadata.Table.getSortedPaths
-        fileInfos.sort(null);
+        if (fileInfos.get(0).getPath().getName().matches("\\d+")) {
```

Comment:
Without file renaming, how did we sort bucket file names before? Do we append 0s in front of file names?

Reply:
Yes correct. We append 0s in front.